### DISCUSSION TITLE
Extract temperature field on object face.

### DISCUSSION THREAD

**Discussion by rohskopf** [ansys/pyaedt#4216] [Q&A] [answered]
In AEDT Icepak I want to extract the temperature field on an object face. How to do this in PyAEDT?

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @rohskopf ,

I would use a Face monitor and then get with get_solution_data the temperature. 

https://aedt.docs.pyansys.com/version/0.7/API/_autosummary/pyaedt.icepak.Icepak.assign_surface_monitor.html#assign-surface-monitor

This will return a FaceMonitor object:
https://aedt.docs.pyansys.com/version/0.7/API/_autosummary/pyaedt.modules.monitor_icepak.FaceMonitor.html

If you do not like this approach, you can directly compute the Temperature of the object/Face using the fields calculator:

https://aedt.docs.pyansys.com/version/0.7/API/_autosummary/pyaedt.modules.AdvancedPostProcessing.PostProcessor.get_scalar_field_value.html#get-scalar-field-value

Hopefully it helps you.
